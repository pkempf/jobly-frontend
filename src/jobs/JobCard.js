import React from "react";
import Card from "react-bootstrap/Card";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import Button from "react-bootstrap/Button";

const JobCard = ({ job, apply }) => {
  return (
    <Card className="CompanyCard mb-2">
      <Card.Body>
        <Card.Title>{job.title}</Card.Title>
        <Card.Subtitle className="mb-2 text-muted">
          {job.companyName}
        </Card.Subtitle>
        <Row>
          <Col sm={9}>
            <span className="font-weight-light">Salary: {job.salary || 0}</span>
            <br />
            <span className="font-weight-light">Equity: {job.equity || 0}</span>
          </Col>
          <Col sm={3}>
            <div className="d-flex justify-content-end m-0 p-0">
              {job.applied ? (
                <Button variant="danger" disabled>
                  APPLIED
                </Button>
              ) : (
                <Button onClick={apply} variant="danger">
                  APPLY
                </Button>
              )}
            </div>
          </Col>
        </Row>
      </Card.Body>
    </Card>
  );
};

export default JobCard;
