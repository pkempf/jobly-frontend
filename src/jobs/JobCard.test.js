import { render } from "@testing-library/react";
import JobCard from "./JobCard";

it("renders without crashing", () => {
  render(<JobCard job={{}} />);
});

it("matches the snapshot", () => {
  const { asFragment } = render(<JobCard job={{}} />);
  expect(asFragment()).toMatchSnapshot();
});
