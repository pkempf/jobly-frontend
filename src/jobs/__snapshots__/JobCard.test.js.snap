// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="CompanyCard mb-2 card"
  >
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      />
      <div
        class="mb-2 text-muted card-subtitle h6"
      />
      <div
        class="row"
      >
        <div
          class="col-sm-9"
        >
          <span
            class="font-weight-light"
          >
            Salary: 0
          </span>
          <br />
          <span
            class="font-weight-light"
          >
            Equity: 0
          </span>
        </div>
        <div
          class="col-sm-3"
        >
          <div
            class="d-flex justify-content-end m-0 p-0"
          >
            <button
              class="btn btn-danger"
              type="button"
            >
              APPLY
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
